### PR TITLE
Fix failing upgrade command when gRPC server interrupts connection

### PR DESCRIPTION
**Pull request by rdner**
## What does this PR do?

The upgrade CLI command creates a client, connects to the Elastic Agent server and triggers the upgrade process which is supposed to shut down the agent and start the upgrade. Sometimes, this happens before the client gets a successful response and the upgrade command returns an error code although the upgrade succeeded.

This change checks if the connection was interrupted and treats it as a successful result instead of error.

## Why is it important?

This race condition occasionally fails our integration tests.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
~~- [ ] I have added an integration test or an E2E test~~

## Related issues

- Fixes https://github.com/elastic/elastic-agent/issues/3890

### CODE REVIEW

File: testing/upgradetest/upgrader.go [hunk 1]
```diff
@@ -325,7 +328,15 @@ func PerformUpgrade(
 
 	upgradeOutput, err := startFixture.Exec(ctx, upgradeCmdArgs)
 	if err != nil {
-		return fmt.Errorf("failed to start agent upgrade to version %q: %w\n%s", endVersionInfo.Binary.Version, err, upgradeOutput)
+		s, ok := status.FromError(err)
```
Comment (blakerouse) [inline testing/upgradetest/upgrader.go:331]: Does this really work here? The output from `startFixture.Exec` is an error from a command output and not directly from the GRPC client as the code above is.

Comment (blakerouse) [reply testing/upgradetest/upgrader.go:331]: Wow really from string output? That is surprising, but awesome if that works!

Comment (rdner) [reply testing/upgradetest/upgrader.go]: Fixed in https://github.com/elastic/elastic-agent/pull/4519/commits/1b17b6da507e7b170e9686f9be14b012d21f19e1

**Review by blakerouse — APPROVED** [review]
Looks good, thanks for the fix!

File: testing/upgradetest/upgrader.go (hunk 1 above)
Comment (rdner) [reply testing/upgradetest/upgrader.go]: Added a test in https://github.com/elastic/elastic-agent/pull/4519/commits/5cb86cd6016a9e100705f8b493ea9cf796b55b22

**Comment by rdner** [comment]
I still need to fix the new test for Windows.

**Pull request merged**